packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart

```diff
@@ -0,0 +1,229 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'types/types.dart';
+
+/// The interface that implementations of in_app_purchase must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `in_app_purchase`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [InAppPurchasePlatform] methods.
+abstract class InAppPurchasePlatform extends PlatformInterface {
+  /// Constructs a InAppPurchasePlatform.
+  InAppPurchasePlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static InAppPurchasePlatform? _instance;
+
+  /// The default instance of [InAppPurchasePlatform] to use.
+  static InAppPurchasePlatform get instance {
+    final InAppPurchasePlatform? platform = _instance;
+    if (platform == null) {
+      throw UnimplementedError(
+          'No platform specific implementation set. Please make sure you set the `instance` with a valid platform specific implementation of the `InAppPurchasePlatform` class.');
+    }
+
+    return platform;
+  }
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [InAppPurchasePlatform] when they register themselves.
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(InAppPurchasePlatform instance) {
```

Comment:
If we don't need the `_instance`, maybe we can create a method like:
```
static void verifyToken(InAppPurchasePlatform instance)  {
    PlatformInterface.verifyToken(instance, _token);
}
```

And we call this method in the instance setter in `InAppPurchaseConnection`

Reply:
Yes I think that would be a good solution.

Reply:
done